src/sparseml/version.py

```diff
@@ -16,16 +16,34 @@
 Functionality for storing and setting the version info for SparseML
 """
 
+from datetime import date
+
+
+_BASE_VERSION = "0.2.0"
+_NIGHTLY = True  # change to False to turn off the build version for releases
```

Comment:
if this is only used with `make build`, can we propagate the nightly build param here?

Reply:
Similar to Ben's question, currently we build nightly with following command:
    make build BUILD_ARGS=nightly

Is the change here to remove the need of BUILD_ARGS=nightly and only relying on changing _NIGHTLY in the file? I'd prefer keeping the old way though so it'll be able to build both regular and nightly builds without changing any code.

Reply:
The nightly is still needed in the setup.py because we change the package name based on that. 

This file is compiled into the source code as-is and therefore cannot rely on build args. A user running the code locally wouldn't have that nightly flag set for any of the code running and we would create the wrong version. In that way, we need to change this file in the source code that goes into the package so it matches what is distributed as the nightly version. The only two ways I think we have for going this is either changing it in git for the release branches or having jenkins override before it creates the package.

Another benefit of setting this arg to True by default is that if a user runs into a problem locally we know it's local or nightly and not on a release by checking the version string. 

We can rename the arg if we feel there is a more appropriate name.

Reply:
So if I understand correctly, after this change:
  ==>to build nightly builds: 
               make build BUILD_ARGS=nightly
  ==> to build regular builds:
               modify _NIGHTLY to False in version.py
               make build
Is this correct?

Reply:
@dhuangnm I pushed up another change to make this step a little bit smoother and remove nightly altogether. Now you can set is_release in version.py to True for release builds and set to False for nightly (the default)